### ISSUE TITLE
using multipage angular-ui-tour with ui-router

### ISSUE THREAD

**Issue by Seshagiri25** [benmarch/angular-ui-tour#29]
Hi,
  Thanks for the awesome library...
With regards to my question, I am using ui-router for navigating between views. To continue on next page I have to set tour-step-next-path="path/to/view", otherwise the next button is hidden on the tour step. But when I set tour-step-next-path, it won't redirect to appropriate view. I have even tried tour-step-on-next="onNext()" and called $state.transitionTo("state") inside controller but still it won't redirect.  Could you help me please to resolve this issue.

Thanks...

**Comment by benmarch** [comment]
Hey @Seshagiri25, the next/prev-path attributes unfortunately won't work with UI Router, but there may be a way to handle it yourself. Try something like this in your onNext() function:

```js
scope.onNext = function (tour) {
  return tour.end().then(function () {
    return $state.transitionTo('state');
  }).then(function () {
    return tour.startAt('nextStepId');
  });
};
```

All tour methods return a promise, and onNext will wait for the promise to resolve before displaying anything else. UI Router's transitionTo() method also returns a promise, so that fits into the chain nicely. I think the above code will work (you may need to use the uiTourService to get a reference to the tour depending on how your scopes are set up) but if not, I can try to build UI Router integration into the next release. Let me know how this goes.

Thanks,
Ben

**Comment by Seshagiri25** [comment]
Hi Ben,
  Thank you very much... It worked great... Don't know why, it gave a method not found error for  startAt, But pause() and resume() worked great...

**Comment by Seshagiri25** [comment]
Although It would be nice if UI Router is integrated... It will be helpful for lot of guys who use UI Router for navigation...

**Comment by benmarch** [comment]
Ah cool, I wasn't sure if pause and resume would work. startAt was released in the latest version (0.4.0) maybe you are using an earlier one? Yeah I've added UI Router integration to the next release (aiming for next week) so I'll let you know if I am able to do that. Thanks for the suggestion!

**Comment by benmarch** [comment]
Hey @Seshagiri25, I have added UI Router integration to the next release (0.5.0), and tagged a pre-release for testing (0.5.0-rc.1). Would you mind installing the pre-release and testing this feature for me? It _should_ work exactly like the nextPath/prevPath works currently, except that you can use the name of a state instead of a path for those options. Just set the "useUiRouter" option to true for the tour, and set up the nextPath/prevPath, and nextStep/prevStep options and it should just work. (This is all in the readme now also.)

Thanks!
Ben

**Comment by benmarch** [comment]
Released in 0.5.0.

**Comment by ddemid** [comment]
Hello,

I would suggest using `$state.go('state');` instead of `$state.transitionTo('state');`
As transitionTo a low level version which requires passing options in most of the cases.
